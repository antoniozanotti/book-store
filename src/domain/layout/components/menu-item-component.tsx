import { MenuItemComponentInterface } from "./menu-item-component-interface";
import Link from "next/link";

export function MenuItemComponent({
  href,
  title,
  children,
  ...props
}: MenuItemComponentInterface) {
  const liStyles = "bg-primary-300 hover:bg-primary-400";
  props.className = props.className
    ? `${liStyles} ${props.className}`
    : liStyles;
    
  return (
    <li {...props}>
      {href ? (
        <Link href={href} title={title} className="p-4 lg:px-10 lg:py-2 block">
          {children ?? title}
        </Link>
      ) : (
        children ?? title
      )}
    </li>
  );
}
