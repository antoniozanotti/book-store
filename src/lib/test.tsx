import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
