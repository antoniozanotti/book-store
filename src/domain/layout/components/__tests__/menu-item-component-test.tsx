import { render, screen } from "@/lib/test";
import { MenuItemComponent } from "../menu-item-component";

describe("MenuItemComponent", () => {
  it("should render without link", async () => {
    const { container } = render(<MenuItemComponent>Test</MenuItemComponent>);
    expect(container).toMatchSnapshot();
  });

  it("should render with link", async () => {
    const { container } = render(<MenuItemComponent href="test" title="test">Test</MenuItemComponent>);
    expect(container).toMatchSnapshot();
  });
});
