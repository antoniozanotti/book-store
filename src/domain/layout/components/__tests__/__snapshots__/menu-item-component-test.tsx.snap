// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItemComponent should render with link 1`] = `
<div>
  <div
    data-testid="container"
  >
    <li
      class="bg-primary-300 hover:bg-primary-400"
    >
      <a
        class="p-4 lg:px-10 lg:py-2 block"
        href="test"
        title="test"
      >
        Test
      </a>
    </li>
  </div>
</div>
`;

exports[`MenuItemComponent should render without link 1`] = `
<div>
  <div
    data-testid="container"
  >
    <li
      class="bg-primary-300 hover:bg-primary-400"
    >
      Test
    </li>
  </div>
</div>
`;
